### DISCUSSION TITLE
Import code using $lib/ (SvelteKit)

### DISCUSSION THREAD

**Discussion by cellulosa** [triggerdotdev/trigger.dev#1247] [Show and tell]
In case it could be useful to others, here's how I managed to make trigger.dev to use SvelteKit's $lib definition

```ts
// ./trigger.config.ts

export const config: TriggerConfig = {
	(...)
	tsconfigPath: './src/lib/server/trigger/tsconfig.json'
};
```

```json
// ./src/lib/server/trigger/tsconfig.json

{
	"extends": ["../../../../tsconfig.json"],
	"compilerOptions": {
		"baseUrl": "../../../../",
		"paths": {
			"$lib": ["src/lib"],
			"$lib/*": ["src/lib/*"]
		}
	}
}
```

that's it, now on your triggers you can import other files as usual e.g.

```ts
// /src/lib/server/trigger/example.ts

import { myFunction } from '$lib/server/myFunction';
```

Perhaps someone will come up with a better approach, as currently one has to define $lib (or any other) manually...

**Comment by perostlin** [comment]
Hello @cellulosa,

Nice tip!

I do have a question regarding the $env in SvelteKit, have you managed to get this to work?
When I have a client towards supabase setup as code below, I get an error saying:
```Error: Worker failed to start Error: Cannot find module '$env/dynamic/private'```

```typescript
import { env } from '$env/dynamic/private';
import { createClient } from '@supabase/supabase-js';

export const supabase = createClient(env.SUPABASE_URL, env.SUPABASE_SERVICE_ROLE_KEY);

```

Thankful for any tips regarding this 🙇‍♂️

**Reply by cellulosa** [reply]
Unfortunately I couldn't get that working, so reverted to using [dotenv](https://www.npmjs.com/package/dotenv):

```ts
import 'dotenv/config';

export const config: TriggerConfig = {
	project: process.env.TRIGGER_PROJECT_ID!,
```

**Reply by perostlin** [reply]
Alright, will do that to, thanks for the quick response!
